File: ios/ParselyReactNativeSdk.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ParselyReactNativeSdk, NSObject)

RCT_EXTERN_METHOD(configure:(NSString *)siteId)

RCT_EXTERN_METHOD(trackPageView: (NSString *)url urlref:(NSString *)urlref extraData:(NSDictionary *)extraData siteId:(NSString *)siteId canonical_url:(NSString *)canonical_url pub_date:(NSString *)pub_date title:(NSString *)title authors:(NSArray *)authors image_url:(NSString *)image_url section:(NSString *)section tags:(NSArray *)tags)

RCT_EXTERN_METHOD(startEngagement:(NSString *)url urlref:(NSString *)urlref extraData:(NSDictionary *)extraData siteId:(NSString *)siteId)

RCT_EXTERN_METHOD(stopEngagement)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/ParselyReactNativeSdk.swift
import ParselyAnalytics

@objc(ParselyReactNativeSdk)
class ParselyReactNativeSdk: NSObject {

    @objc func configure(_ siteId: String) -> Void {
        Parsely.sharedInstance.configure(siteId: siteId)
    }
    
    @objc func trackPageView(_ url: String, urlref: String, extraData: Dictionary<String, Any>, siteId: String, canonical_url: String, pub_date: String, title: String, authors: Array<String>, image_url: String, section: String, tags: Array<String>) -> Void {
        let dateFormatter = ISO8601DateFormatter()
        dateFormatter.formatOptions = [
            .withFullDate,
            .withFullTime,
            .withDashSeparatorInDate,
            .withFractionalSeconds]
        let date = dateFormatter.date(from: pub_date)
        let metadata = ParselyMetadata(canonical_url: canonical_url, pub_date: date, title: title, authors: authors, image_url: image_url, section: section, tags: tags)
        Parsely.sharedInstance.trackPageView(url: url, urlref: urlref, metadata: metadata, extraData: extraData, siteId: siteId)
    }

    @objc func startEngagement(_ url: String, urlref: String, extraData: Dictionary<String, Any>, siteId: String) -> Void {
        Parsely.sharedInstance.startEngagement(url: url, urlref: urlref, extraData: extraData, siteId: siteId)
    }

    @objc func stopEngagement() -> Void {
        Parsely.sharedInstance.stopEngagement()
    }
}
